modules/ml/src/main/java/org/apache/ignite/ml/trainers/DatasetTrainer.java

```diff
@@ -85,7 +85,36 @@
      * @param <V> Type of a value in {@code upstream} data.
      * @return Model.
      */
-    public abstract <K, V> M fit(DatasetBuilder<K, V> datasetBuilder, Preprocessor<K, V> preprocessor);
+    public <K, V> M fit(DatasetBuilder<K, V> datasetBuilder, Preprocessor<K, V> preprocessor) {
+        learningEnvironment().deployContext().init(preprocessor);
```

Comment:
Does it make sense to add method `initDeployContext` into the learning environment and don't expose deploy context object itself?